docs/shared-content-source/docs/modules/develop/pages/schema-first-approach.adoc

```diff
@@ -111,6 +111,21 @@ Here `CustomHandler` is any function adhering to the following interface
 errorHandler: (Array[Byte], Throwable) => Option[T]
 ----
 
+== ByteArray Inlets and outlets
+
+As mentioned earlier, Cloudflow uses a schema-first approach for building streaming data pipelines.
+This works great for intra streamlet communications, but often falls short for integration
+with existing applications. Cloudflow currently supports Avro and Protobuf schemas, which might be an
+issue in cases, when existing applications do not support either one of them. ByteArray inlets and outlets
+are introduced specifically to support such integrations.
```

Comment:
I think we should add some docs on how to write your own codecs. ByteArray is a fallback / catch all, but in many cases I would assume users don't actually want to read/write byte arrays, they want to use their own codecs, for "external use". maybe have some `ExternalInlet` / `ExternalOutlet` that can only be attached on one side of a topic (producer or consumer), and exclude these in verification of the blueprint. LMKWYT.

Reply:
As I was writing in the doc, I do not see this used for streamlet to streamlet communications. Its a way to integrate with other applications, that do not support Avro or Proto. I am not sure we should encourage writing custom codecs, especially for streamlet to streamlet. Its just a connivence method for custom integrations

Reply:
no I'm saying specifically for external communications.

Reply:
`ExternalInlet`, `ExternalOutlet`, make that formal.

Reply:
But we can do that , if users ask for it, in a separate PR. Right now this is good enough.